### DISCUSSION TITLE
Loading X509 Certificate in Ubuntu

### DISCUSSION THREAD

**Discussion by ryanbuening** [ravendb/ravendb#13827] [Q&A] [answered]
I'm running a .NET Core app in Ubuntu and trying to connect to a cloud RavenDB instance with the following:

```c#
var bytes = File.ReadAllBytes("/etc/ssl/certs/foo.pfx");
var clientCert = new X509Certificate2(bytes, "foo");

var store = new DocumentStore
{
	Urls = new[] { "https://a.free.foo.ravendb.cloud" },
	Certificate = clientCert,
	Database = "FooDatabase"
};

store.Initialize();
builder.Services.AddSingleton<IDocumentStore>(store);
```

When the app starts, I receive the following error:

System.Security.Cryptography.CryptographicException: ASN1 corrupted data.
 ---> System.Formats.Asn1.AsnContentException: The encoded length exceeds the maximum supported by this library (Int32.MaxValue).
   at System.Formats.Asn1.AsnDecoder.ReadLength(ReadOnlySpan`1 source, AsnEncodingRules ruleSet, Int32& bytesConsumed)
   at System.Formats.Asn1.AsnDecoder.ReadEncodedValue(ReadOnlySpan`1 source, AsnEncodingRules ruleSet, Int32& contentOffset, Int32& contentLength, Int32& bytesConsumed)
   at Internal.Cryptography.Pal.UnixPkcs12Reader.ParsePkcs12(ReadOnlySpan`1 data)

Certs in unix are new to me. Are there any docs or references I could look at to get a better idea?

**Comment by ppekrol** [comment, marked as answer]
Hi @ryanbuening 

I'm assuming that the foo.pfx is a client certificate generated by RavenDB?

Have you tried opening it as follows:
```
X509Certificate2 clientCertificate = new X509Certificate2("/etc/ssl/certs/foo.pfx", "foo");
```

**Reply by ryanbuening** [reply]
It looks like my issue was my pfx file was corrupted somehow. Probably due to the way I tried to copy it to the server. Your syntax did work though after I replaced my pfx file so I don't need to call `ReadAllBytes`. Thanks!
